File: factory/typescript/concreteCreatorSensors/createPresionSensor.ts
/** @format */

import { Creator } from "../creatorFactory";
import { SensorOperations } from "../gateway";
import { Presion } from "../concreteSensors/presion";

export class CreatePresionSensor implements Creator {
  public info(): void {
    console.log("presion works!!")
  }
  public factoryMethod(): SensorOperations {
    return new Presion();
  }
}

File: factory/python/concreteCreatorSensors/createPresionSensor.py

from concreteSensors.presion import Presion
from gateway import SensorOperations
from creatorFactory import Creator

class CreatePresionSensor(Creator):

    def info(self) -> None:
        print("presion works!!")

    def factoryMethod(self) -> SensorOperations:
        return Presion()

